pkg/operator/operator/validations.go

```diff
@@ -744,3 +789,54 @@ func getValidationK8sResources() ([]kunstructured.Unstructured, *kresource.Quant
 
 	return virtualServices, maxMem, err
 }
+
+func validateDockerImagePath(image string) error {
+	if _, err := cr.ValidateImageVersion(image); err != nil {
+		return err
+	}
+
+	dockerAuth := dockerlib.NoAuth
+	if regex.IsValidECRURL(image) {
+		operatorID, _, err := config.AWS.GetCachedAccountID()
+		if err != nil {
+			return err
+		}
+		registryID := aws.GetAccountIDFromECRURL(image)
+
+		if strings.Compare(operatorID, registryID) != 0 {
```

Comment:
nit: `if operatorID != registryID` is simpler

Reply:
Agreed. Done!